packages/mattermost-redux/src/store/helpers.ts

```diff
@@ -1,28 +1,50 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import {combineReducers} from 'redux';
+import {combineReducers, AnyAction} from 'redux';
 
 import {General} from '../constants';
 
 import {enableBatching, Action, Reducer} from 'mattermost-redux/types/actions';
+import {GlobalState} from 'mattermost-redux/types/store';
+import deepFreezeAndThrowOnMutation from 'mattermost-redux/utils/deep_freeze';
 
 import reducerRegistry from './reducer_registry';
 
-export function createReducer(baseState: any, ...reducers: Reducer[]) {
+export function createReducer(baseState: GlobalState, ...reducers: Reducer[]): Reducer<GlobalState, Action> {
     reducerRegistry.setReducers(Object.assign({}, ...reducers));
     const baseReducer = combineReducers(reducerRegistry.getReducers());
 
     // Root reducer wrapper that listens for reset events.
     // Returns whatever is passed for the data property
     // as the new state.
-    function offlineReducer(state = {}, action: Action) {
+    const offlineReducer = (state: GlobalState, action: Action) => {
         if ('type' in action && 'data' in action && action.type === General.OFFLINE_STORE_RESET) {
```

Comment:
At some point, it might be worth investigating how much of this is even needed any more. A lot of it was ported over from mobile where there was more logic required for the offline storage of stuff, and I know I cleaned up a lot of dead code when moving mattermost-redux into the web app. `OFFLINE_STORE_RESET` is referenced in the webapp's store initialization, but I don't know if that branch actually triggers at all.